### PR TITLE
Verify SHC signature based on elliptic curve(required) instead of x5c(optional)

### PR DESCRIPTION
**Pull request by NC-204**

### CODE REVIEW

File: FHICORC.Core/Services/Model/Exceptions/InvalidSignatureException.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace FHICORC.Core.Services.Model.Exceptions
+{
+    /// <summary>
+    ///  Exception for invalid signature
+    /// </summary>
+    public class InvalidSignatureException : System.Exception
```
Comment (NC-395) [inline FHICORC.Core/Services/Model/Exceptions/InvalidSignatureException.cs:6]: Import System with `using` statement

File: FHICORC.Core/Services/DecoderService/CertificationService.cs [hunk 2]
```diff
@@ -113,45 +109,42 @@ private string GetJsonValue(string json, string jsonKey)
...
-                var signer = SignerUtilities.GetSigner("SHA-256withPLAIN-ECDSA");
+            byte[] MessageBytes = Encoding.UTF8.GetBytes(jws.EncodedPart(JwtPartsIndex.Header) + '.' + jws.EncodedPart(JwtPartsIndex.Payload));
+            byte[] Signature = jws.DecodedSignature();
 
-                signer.Init(false, publicKey);
-                signer.BlockUpdate(msgBytes, 0, msgBytes.Length);
-                return signer.VerifySignature(sigBytes);
-            }
-            catch (Exception exc)
+            var signer = SignerUtilities.GetSigner(BsiObjectIdentifiers.ecdsa_plain_SHA256);
```
Comment (NC-395) [inline FHICORC.Core/Services/DecoderService/CertificationService.cs:138]: Declare type explicitly

File: FHICORC.Core/Services/DecoderService/CertificationService.cs [hunk 3]
```diff
@@ -113,45 +109,42 @@ private string GetJsonValue(string json, string jsonKey)
...
+            signer.Init(false, jwkSignature);
+            signer.BlockUpdate(MessageBytes, 0, MessageBytes.Length);
+            var isSignatureValid = signer.VerifySignature(Signature);
```
Comment (NC-395) [inline FHICORC.Core/Services/DecoderService/CertificationService.cs:141]: Declare type explicitly

**Review by NC-395 — CHANGES_REQUESTED** [review]
Very nice, just small things to fix

**Comment by NC-204** [comment]
@NC-395  fixed feedback 🙂

**Review by NC-395 — APPROVED** [review]

**Pull request merged**